### PR TITLE
feat: change address screening service to Uniswap /screen API

### PR DESCRIPTION
**Pull request by kristiehuang**
## Description
* Change our address screening service from current cloudflare worker ([wallet-screening-server](https://github.com/Uniswap/wallet-screening-server)) to Uniswap /screen API

_Linear ticket:_ https://linear.app/uniswap/issue/WEB-2587/move-web-app-trm-checking-to-use-same-backend-endpoint-as-the-mobile
_PR to API repo:_ https://github.com/Uniswap/api/pull/110

## Test plan
* unit tests in /interface repo

### QA (ie manual testing)
* replace UNISWAP_API_V1 in .env using my deployed AWS dev API (https://xtr344xw65.execute-api.us-east-1.amazonaws.com/prod/screen)

### CODE REVIEW

File: src/hooks/useAccountRiskCheck.test.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import { ApplicationModal, setOpenModal } from 'state/application/reducer'
```
Comment (JFrankfurt) [inline src/hooks/useAccountRiskCheck.test.ts:1]: nit: prefer unit testing the component that uses the hook

File: src/hooks/useAccountRiskCheck.ts [hunk 2]
```diff
@@ -11,11 +11,14 @@ export default function useAccountRiskCheck(account: string | null | undefined)
...
       try {
+        // Check local browser cache
         const storedTime = localStorage.getItem(riskCheckLocalStorageKey)
         const checkExpirationTime = storedTime ? parseInt(storedTime) : now - 1
         if (checkExpirationTime < Date.now()) {
-          const headers = new Headers({ 'Content-Type': 'application/json' })
-          fetch('https://screening-worker.uniswap.workers.dev', {
+          const headers = new Headers({
+            'Content-Type': 'application/json',
+          })
```
Comment (JFrankfurt) [inline src/hooks/useAccountRiskCheck.ts:20]: odd that this formatting is changing

Comment (JFrankfurt) [reply src/hooks/useAccountRiskCheck.ts:20]: you might want to check that your editor is correctly configured to use our lint/prettier setup

File: src/hooks/useAccountRiskCheck.ts [hunk 3]
```diff
@@ -26,9 +29,13 @@ export default function useAccountRiskCheck(account: string | null | undefined)
                 dispatch(setOpenModal(ApplicationModal.BLOCKED_ACCOUNT))
               }
             })
-            .catch(() => dispatch(setOpenModal(null)))
+            .catch((e) => {
+              console.error(e)
```
Comment (JFrankfurt) [inline src/hooks/useAccountRiskCheck.ts:33]: we don't want a console error surfaced here

File: src/hooks/useAccountRiskCheck.ts [hunk 4]
```diff
@@ -11,11 +11,12 @@ export default function useAccountRiskCheck(account: string | null | undefined)
       const riskCheckLocalStorageKey = `risk-check-${account}`
       const now = Date.now()
       try {
+        // Check local browser cache
         const storedTime = localStorage.getItem(riskCheckLocalStorageKey)
         const checkExpirationTime = storedTime ? parseInt(storedTime) : now - 1
         if (checkExpirationTime < Date.now()) {
           const headers = new Headers({ 'Content-Type': 'application/json' })
-          fetch('https://screening-worker.uniswap.workers.dev', {
+          fetch('https://api.uniswap.org/v1/screen', {
```
Comment (justindomingue) [inline src/hooks/useAccountRiskCheck.ts:19]: is there a reason this isn't using `react-query`?

**Review by justindomingue — APPROVED** [review]

File: src/hooks/useAccountRiskCheck.ts (hunk 4 above)
Comment (tinaszheng) [reply src/hooks/useAccountRiskCheck.ts:19]: it probably should but then again... the added middleware overhead will be annoying xD

**Pull request merged**